### PR TITLE
(DOCSP-33483): Swift: Actor and Concurrency updates and fixes

### PR DESCRIPTION
**Pull request by dacharyc**
## Pull Request Info

This PR:

- Fixes and un-skips failing tests showing invalid examples of using Realm with Swift Actors
- Corrects `writeAsync` mentions to `asyncWrite`
- Corrects the conflated "Observe Notifications on an Actor-Isolated Realm" to "Observe Notifications on a Different Actor"
- Adds more details about how to use Realm across actors based on docs feedback
- Notes you can't currently use `thaw()` with Swift actors, and recommends `ThreadSafeReference` instead

### Jira

- https://jira.mongodb.org/browse/DOCSP-33483
- https://jira.mongodb.org/browse/DOCSP-34052
- https://jira.mongodb.org/browse/DOCSP-33834

### Staged Changes

- [Use Realm with Actors](https://preview-mongodbdacharyc.gatsbyjs.io/realm/DOCSP-33483/sdk/swift/use-realm-with-actors/) (formerly Actor-Isolated Realms - but that was conflating some of the functionality)
- [Swift Concurrency](https://preview-mongodbdacharyc.gatsbyjs.io/realm/DOCSP-33483/sdk/swift/swift-concurrency/)
- [Threading](https://preview-mongodbdacharyc.gatsbyjs.io/realm/DOCSP-33483/sdk/swift/crud/threading/)
- [React to Changes](https://preview-mongodbdacharyc.gatsbyjs.io/realm/DOCSP-33483/sdk/swift/react-to-changes/#react-to-changes-on-a-different-actor)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/examples/generated/code/start/RealmActor.snippet.observe-collection-on-actor.swift [hunk 1]
```diff
@@ -1,36 +1,65 @@
...
-// Invalidate the token when done observing
-token.invalidate()
+// Execute some code on a different actor - in this case, the MainActor
+@MainActor
+func mainThreadFunction() async throws {
+    let realm = try! await Realm()
+    
+    // Create a todo item so there is something to observe
+    try await realm.asyncWrite {
+        return realm.create(Todo.self, value: [
```
Comment (tgoyne) [inline source/examples/generated/code/start/RealmActor.snippet.observe-collection-on-actor.swift:23]: nit: `return` isn't doing anything here

File: source/examples/generated/code/start/RealmActor.snippet.observe-collection-on-actor.swift [hunk 2]
```diff
@@ -1,36 +1,65 @@
...
-            print("An object was deleted: \(changes)")
-        } else if !insertions.isEmpty {
-            print("An object was inserted: \(changes)")
-        } else if !modifications.isEmpty {
-            print("An object was modified: \(changes)")
+// Create a simple actor
+@globalActor actor BackgroundActor: GlobalActor {
+    static var shared = BackgroundActor()
+    
+    public func deleteTodo(tsrToTodo tsr: ThreadSafeReference<Todo>) throws {
```
Comment (tgoyne) [inline source/examples/generated/code/start/RealmActor.snippet.observe-collection-on-actor.swift:5]: While you can put functions on a global actor, it'd be more common to have this be a top-level `@BackgroudActor func`, and would serve better as documentation for global actor vs regular actor.

**Review by tgoyne — APPROVED** [review]

File: source/sdk/swift/crud/threading.txt [hunk 3]
```diff
@@ -37,7 +37,7 @@ simplify this for you.
    <ios-thread-safe-reference>` or :ref:`frozen objects <ios-frozen-objects>` 
    across threads.
 
-Actor-Isolated Realms
+Use Realm with Actors
 ---------------------
 
 This page describes how to manually manage realm files and objects across threads.
```
Comment (cbullinger) [inline source/sdk/swift/crud/threading.txt:43]: maybe there's a good reason for having it here, but it was confusing to me to have this as a section heading that also links out to a page with the same name. Maybe remove this heading and move this section to the top of the page as the intro?

File: source/sdk/swift/crud/threading.txt [hunk 4]
```diff
@@ -399,6 +401,13 @@ to other threads. You can freely share the frozen object across threads
 without concern for thread issues. When you freeze a realm, its child 
 objects also become frozen.
 
+.. tip:: Use ThreadSafeReference with Swift Actors
+
+   Realm does not currently support using ``thaw()`` with Swift Actors.
```
Comment (cbullinger) [inline source/sdk/swift/crud/threading.txt:406]:
```suggestion
   Realm does not currently support using ``thaw()`` with Swift actors.
```
or is there a capitalization rule i'm not following?

File: source/sdk/swift/crud/threading.txt [hunk 5]
```diff
@@ -90,11 +90,11 @@ Perform a Background Write
 .. versionadded:: 10.26.0
 
 You can add, modify, or delete objects in the background using 
-:swift-sdk:`writeAsync <Structs/Realm.html#/s:10RealmSwift0A0V10writeAsync_10onCompletes6UInt32Vyyc_ys5Error_pSgcSgtF>`. 
+:swift-sdk:`asyncWrite <Structs/Realm.html#/s:10RealmSwift0A0V10writeAsync_10onCompletes6UInt32Vyyc_ys5Error_pSgcSgtF>`.
```
Comment (cbullinger) [inline source/sdk/swift/crud/threading.txt:93]: this links out to `writeAsync(_:onComplete:)`. is that correct?

File: source/sdk/swift/crud/threading.txt [hunk 6]
```diff
@@ -207,9 +207,10 @@ on your use case:
 - To keep and share many read-only views of the object in your app, copy
   the object from the realm.
 
-- To share an instance of a realm or specific object with another thread, share
-  a :ref:`thread_safe_reference <ios-thread-safe-reference>` to the realm
-  instance or object.
+- To share an instance of a realm or specific object with another thread or
+  across actor boundaries, share a :ref:`thread_safe_reference
```
Comment (cbullinger) [inline source/sdk/swift/crud/threading.txt:211]:
```suggestion
  across actor boundaries, share a :ref:`thread-safe reference 
```

**Review by cbullinger — APPROVED** [review]
Some random Q's from me, but overall LGTM

File: source/sdk/swift/react-to-changes.txt [hunk 7]
```diff
@@ -322,21 +322,21 @@ does support this for code compatibility.
    Examples of using Realm with :github:`ReactiveCocoa from Objective-C<realm/realm-swift/tree/master/examples/ios/objc/TableView>`, 
    and :github:`ReactKit from Swift<realm/realm-swift/tree/v2.3.0/examples/ios/swift-2.2/ReactKit>`.
 
-.. _swift-react-to-changes-actor-confined-realm:
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/sdk/swift/react-to-changes.txt:325]: I wasn't actually linking to this anywhere, so it was safe to remove it. It's a good thing to flag, though!

File: source/examples/generated/code/start/RealmActor.snippet.observe-collection-on-actor.swift (hunk 2 above)
Comment (dacharyc) [reply source/examples/generated/code/start/RealmActor.snippet.observe-collection-on-actor.swift:3]: Ok, I've simplified a few of these examples to remove the global actor aspect. Is this more in line with what you had in mind, @tgoyne ?

**Pull request merged**